### PR TITLE
Add unit tests for validators

### PR DESCRIPTION
**Pull request by pyasi**
### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Added tests for the ValidateAPIKey class. A previously untested piece of functionality
- Made small changes in the ValidateAPIKey class to conform to project standards

### CODE REVIEW

File: test/test_validators.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    import unittest
+
+
+class TestValidateAPIKey(unittest.TestCase):
+
+    def test_basic_validator_initialization(self):
+        validator = ValidateAPIKey()
+        self.assertIsInstance(validator.regexes, set)
+        self.assertTrue(validator.regexes)
+        assert len(validator.regexes) == 1
```
Comment (42B) [inline test/test_validators.py]: `self.assertEqual(len(validator.regexes), 1)`?

* https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

File: test/test_validators.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_validator_with_own_regexes(self):
+        validator = ValidateAPIKey(regex_strings=['test_one', 'test_two'])
+        self.assertTrue(validator.regexes)
+        assert len(validator.regexes) == 3
```
Comment (42B) [inline test/test_validators.py]: `self.assertEqual(len(validator.regexes), 3)`?

* https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

File: test/test_validators.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_validator_no_default(self):
+        validator = ValidateAPIKey(use_default=False)
+        self.assertFalse(validator.regexes)
+        assert len(validator.regexes) == 0
```
Comment (42B) [inline test/test_validators.py]: `self.assertEqual(len(validator.regexes), 0)`?

* https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

File: test/test_validators.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_can_add_regexes(self):
+        validator = ValidateAPIKey(use_default=False)
+        validator.regexes.add('test')
+        self.assertIsInstance(validator.regexes, set)
+        self.assertTrue(validator.regexes)
+        assert len(validator.regexes) == 1
```
Comment (42B) [inline test/test_validators.py]: `self.assertEqual(len(validator.regexes), 1)`?

* https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

File: test/test_validators.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        validator = ValidateAPIKey(use_default=False)
+        self.assertEqual(len(validator.regexes), 0)
+
+    def test_can_add_regexes(self):
+        validator = ValidateAPIKey(use_default=False)
+        validator.regexes.add('test')
+        self.assertIsInstance(validator.regexes, set)
+        self.assertEqual(len(validator.regexes), 1)
+
+
```
Comment (42B) [inline test/test_validators.py]: Extra blank line?

> *Method definitions inside a class are surrounded by a single blank line.*

* https://pep8.org/#blank-lines

**Comment by pyasi** [comment]
@misterdorm When i rebase against `v4` I get a ton of conflicts even though this branch is working on an isolated piece

**Comment by misterdorm** [comment]
@pyasi Yeah, because your original branch contains all the history from master (since it was branched from there.)  Probably the easiest is to start with a fresh branch, created from v4, and then apply your changes/commits on top of that.  There's not really a great way to remove all the history from master from your original branch.  Sorry this is kind of a hassle.

**Comment by misterdorm** [comment]
Replaced by #729

**Pull request closed without merging**